openlibrary/macros/LoanStatus.html

```diff
@@ -17,7 +17,7 @@
 $ work_key = work_key or (doc.get('works') and doc.works[0].key)
 
 $ waiting_loan_start_time = time()
-$ waiting_loan = check_loan_status and ocaid and ctx.user and ctx.user.get_waiting_loan_for(ocaid)
+$ waiting_loan = check_loan_status and ocaid and ctx.user and ctx.user.get_waiting_loan_for(ocaid, use_cache=True)
```

Comment:
We should manually test some of these memcache deletes (and ensure these values are being deleted across all memcache instances) -- ol-mem[0,1,2]

Reply:
On line 20 here, what we probably want to do is fetch **all** of a patron's waiting loans at once?

There are 2 scenarios:
1. Every time we're on a new book, trying to get the waiting loan and caching the result per each user & ocaid (this still means we're making lots of calls and caching them independently)
2. In LoanStatus, we make a single bulk call for all a patron's holds (waiting list books) and re-cache this every ~10 minutes. So we'd be caching per user 

@jimchamp and I think the 2nd option may be the right way to go.

def get_holds(user_key, optional_ocaid=None):
    user_holds = cache.memoize(
        engine="memcache", key=f"holds_{user_key}", expires=10 * dateutil.MINUTE_SECS)(user_key)
    if optional_ocaid:
        # check if option_ocaid in user_holds
        return next([hold for hold in user_holds if hold.ocaid == optional_ocaid] # psuedo code 
    return user_holds